### PR TITLE
V0.4 different list orders

### PR DESCRIPTION
**Pull request by mattheuslee**
Comparable interface and compareTo(...) has been removed from the entries as they're now using external comparators that we can change as needed.

@ChuaPingChan please advise if how I implemented the API calls is consistent with what is already there, and whether model.entry.util.Comparators should be in a different folder.

@kevinLamKB please advise if my code for the ListCommand and ListCommandParser is consistent with the rest of `logic`

### CODE REVIEW

**Comment by mattheuslee** [comment]
@CS2103JUN2017-T2/developers Coverage will be back up when I add in the UI tests for this. Will be in a separate PR.

**Review by kevinLamKB — APPROVED** [review]

**Comment by kevinLamKB** [comment]
looks okay to me.
will need to change the prefixes to remove / after im done fixing tokenizer but thats an ez fix

**Review by ChuaPingChan — APPROVED** [review]
@mattheuslee 
Looks fine to me 👍

File: src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                                                   ListCommand.MESSAGE_USAGE));
+        }
+
+        return new ListCommand(flags);
+    }
+
+    private boolean hasValidFlagCombination(String... flags) {
+        ArrayList<String> flagList = new ArrayList<String>(Arrays.asList(flags));
+        // Cannot have any unknown flags
+        if (!Arrays.asList(ListCommand.PREFIX_LIST).containsAll(flagList)) {
```
Comment (kevinLamKB) [inline src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java:41]: Thats pretty neat actually. im gonna steal this lol

Comment (mattheuslee) [reply src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java:41]: not really stealing if it's part of the Java library haha 😆

Comment (kevinLamKB) [reply src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java:41]: haha but i was struggling to go and understand ArgumentMultimap and java stream() from addressbook4 when such a neat answer existed 👍 i like this method to check for invalid flags more

Comment (mattheuslee) [reply src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java:41]: yeah, and it reads much better as well, as far as conveying intent.

**Review by akd13 — APPROVED** [review]

**Review by akd13 — APPROVED** [review]

**Pull request merged**